closure/compiler/closure_js_library.bzl

```diff
@@ -68,6 +68,25 @@ def _impl(ctx):
     args += ["--expect_failure"]
   if ctx.attr.internal_expect_warnings:
     args += ["--expect_warnings"]
+
+  files = [ctx.outputs.provided, ctx.outputs.stderr]
+  if ctx.attr.style and len(ctx.files.srcs) > 0:
+    files.append(ctx.outputs.jsstyle)
+    style_args = [ctx.executable._clangformat.path,
+                  ctx.outputs.jsstyle.path]
+    style_inputs = [ctx.executable._clangformat]
+    for direct_src in ctx.files.srcs:
+      style_args += [direct_src.path]
+      style_inputs.append(direct_src)
+    ctx.action(
```

Comment:
>  do you think it would be better if we launched clang format as a concurrent subprocess of JsChecker?

I'm fine doing it that way if you prefer.

One downside is that such a job would be hidden to "bazel query" and therefore is slightly more magical to new users of rules_closure who are coming from a bazel background.

> we might be able to integrate it into JsChecker's error system.

Are you thinking a file wide fileoverview level @suppress or something more granular (eg: function, line level)?

One downside to non-standard clang-format suppressions is that existing clang-format users (eg: C/C++/Java/Objective-C) won't expect it. 

> Therefore clang format errors could be turned off via suppress rather than a new attribute.

Oh, maybe you are thinking a `closure_js_library` level suppression (vs an in-file suppression)? Is that correct?

If so, couldn't we implement that API in `closure_js_library` either via a conditional ctx.action or a JsChecker subprocess? (Basically eat the clangFormat suppression in `closure_js_library` , do not run the ctx.action for clang-format and do not pass the suppression down to JsChecker).

> How feasible does that sound, based on your understanding of clang format so far?

If you are thinking about avoiding a new `style` attribute and conditionally running clang-format when the suppression is not present (either from a conditional ctx.action or a JsChecker subprocess) this sounds very actionable.

If you are thinking of fine grained in-file suppressions it would potentially be complex. I would need to look at the clang-format source to gain an idea of the complexity.